service/service_construction.go

```diff
@@ -412,14 +429,35 @@ func (s ConstructionService) ConstructionPayloads(
 		return nil, wrapError(errInvalidInput, fmt.Errorf("%s is not a valid address", toAddress))
 	}
 
-	tx := ethtypes.NewTransaction(
-		nonce,
-		ethcommon.HexToAddress(checkTo),
-		amount,
-		gasLimit,
-		gasPrice,
-		transferData,
-	)
+	var tx *ethtypes.Transaction
+	if types.Hash(fromCurrency) == types.Hash(mapper.AvaxCurrency) {
+		transferData := []byte{}
+		tx = ethtypes.NewTransaction(
+			nonce,
+			ethcommon.HexToAddress(checkTo),
+			amount,
+			gasLimit,
+			gasPrice,
+			transferData,
+		)
+	} else {
+		contract, ok := fromCurrency.Metadata[mapper.ContractAddressMetadata]
```

Comment:
`fromCurrency.Metadata[mapper.ContractAddressMetadata].(string)` so you don't need to do `.(string)` on 454. (this returns `!ok` if either the key doesn't exist or it is not a string)